spidermon/contrib/scrapy/pipelines.py

```diff
@@ -136,25 +133,16 @@ def find_validators(self, item):
         return find(item.__class__) or find(Item)
 
     def _convert_item_to_dict(self, item):
-        serialized_json = BytesIO()
-        exporter = JsonLinesItemExporter(serialized_json)
-        exporter.export_item(item)
-        data = json.loads(to_unicode(serialized_json.getvalue(), exporter.encoding))
-        serialized_json.close()
-        return data
+        data = ItemAdapter(item)
+        return data.asdict()
 
     def _add_errors_to_item(self, item, errors):
-        try:
-            if self.errors_field not in item.__class__.fields:
-                item.__class__.fields[self.errors_field] = Field()
-            if self.errors_field not in item._values:
-                item[self.errors_field] = defaultdict(list)
-        except AttributeError:
-            # The item is just a dict object instead of a Scrapy.Item object
-            if self.errors_field not in item:
-                item[self.errors_field] = defaultdict(list)
+        data = ItemAdapter(item)
+        if self.errors_field not in data.keys():
```

Comment:
:lipstick: I wonder if `in data:` would work as well.

Reply:
It would because `ItemAdapter` implements `MutableMapping`. It would also be a tiny bit more efficient if I'm not mistaken.